File: Lista 1/src/Craps.java
import java.util.*;


public class Craps {
    public static void main(String[] args) {
        Random gerador = new Random();
        int ponto = -1, lancamento = 1;

        while(true){
            int dado1 = gerador.nextInt(6 ) + 1;
            int dado2 = gerador.nextInt(6 ) + 1;
            int soma = 0;

            soma = dado1 + dado2;
            if (lancamento == 2){
                System.out.println("Iniciando estágio 2");
            }
            if (lancamento == 1){
                System.out.println("Dado 1: " + dado1 + ", Dado 2: " + dado2);
            }else{
                System.out.println("Jogada " + lancamento + ": " + soma);
            }

            if(lancamento == 1 && soma != 7 && soma > 3 && soma < 11) {
                System.out.println("Ponto: " + soma);
                ponto = soma;
            }

            if((soma == 7 || soma == 11) && lancamento == 1){
                System.out.println("Voce venceu");
                break;
            } else if (lancamento !=1 && soma == ponto) {
                System.out.println("Voce venceu");
                break;
            }

            if(soma == 2 || soma == 3 || soma == 12 || (soma == 7 && lancamento != 1)){
                System.out.println("Você perdeu!!");
                break;
            }
            lancamento++;
        }

    }
}

File: Lista 2/Craps.py
import random

ponto = -1
lancamento = 1

while(1):
    soma = 0
    random.seed()
    dado1 = int(random.randrange(0,7,))
    dado2 = int(random.randrange(0,7,))

    soma = dado1 + dado2
    
    if lancamento == 2:
        print("Iniciando estagio 2")
    if lancamento == 1:
        print(f"Dado 1: {dado1} --- Dado 2: {dado2}")
    else: 
        print(f"Jogada {lancamento}: {soma}")

    if (lancamento == 1 and soma != 7 and soma > 3 and soma < 11):
        ponto = soma
        print(f"Ponto: {ponto}")
    
    if((soma == 7 or soma == 11) and lancamento == 1):
        print("Voce venceu!!!")
        break
    elif (lancamento != 1 and soma == ponto):
        print("Voce venceu!!!")
        break
    
    if (soma == 2 or soma == 3 or soma == 12 or (soma == 7 and lancamento !=1)):
        print("Voce perdeu!!!")
        break
    
    lancamento+=1